### DISCUSSION TITLE
Has the template format been finalized?

### DISCUSSION THREAD

**Discussion by kurt-mueller-osumc** [Azure/azure-dev#1115] [Q&A] [answered]
Hey all,

I recently init'd a new azd template, `Azure-Samples/todo-csharp-cosmos-sql`, and noticed that the `resources.bicep` file had completely been eliminated... which is fine, but I'm trying to pick up best practices and opinions from the azd cli team. Is the template format set in stone or will there be further updates/changes?

Thanks all.

Kurt

**Comment by lechnerc77** [comment, marked as answer]
Hi Kurt, 

I am also observing the journey of the infrastructure setup for several releases. Although not being part of the team, I would assume that the structure has undergone several improvements and I guess will be subject to changes based on feedback and experience also in upcoming releases.

I think there will be no final "perfect" solution, however I think the current one (release 0.4.0) is already really good. 
Concerning the resources.bicep: I think this is subject to your preference to some extent. Keeping everything in main.bicep makes it easier to get an overview, but maybe some complex setups are in favor of the split of the files as it was in previous versions.

Maybe that helps.

Cheers Christian
